src/main/kotlin/it/pagopa/wallet/scheduler/jobs/paymentwallet/OnboardedPaymentWalletJob.kt

```diff
@@ -0,0 +1,113 @@
+                        AuditWallet(
+                            paymentMethodId = it.paymentMethodId,
+                            creationDate = it.creationDate.toString(),
+                            updateDate = it.updateDate.toString(),
+                            applications =
+                                it.applications.map { application ->
+                                    AuditWalletApplication(
+                                        id = application.id,
+                                        creationDate = application.creationDate,
+                                        updateDate = application.updateDate,
+                                        metadata = application.metadata,
+                                        status = application.status
+                                    )
+                                },
+                            details =
+                                it.details?.let { walletDetails ->
+                                    when (walletDetails) {
+                                        is CardDetails ->
+                                            AuditWalletDetails(
+                                                type = "CARDS", // TODO check this value
+                                                cardBrand = walletDetails.brand,
+                                                pspId = null
+                                            )
+                                        is PayPalDetails ->
+                                            AuditWalletDetails(
+                                                type = "PAYPAL", // TODO check this value
+                                                cardBrand = null,
+                                                pspId = walletDetails.pspId
+                                            )
+                                    }
+                                },
+                            status = it.status,
+                            validationOperationId = null, // TODO is ok to pass it as null here?
+                            validationOperationResult = it.validationOperationResult,
+                            validationOperationTimestamp =
+                                it.creationDate
+                                    .toString(), // use the wallet creation date as validation
+                            // operation timestamp
+                            validationErrorCode = it.validationErrorCode,
+                        )
+                )
+            }
+            .flatMap { cdcEventDispatcherService.dispatchEvent(it) }
+            .collectList()
+            .map {
+                it.last().timestamp // <- this is the wallet creation date and can be used to save
+                // it to
+                // Redis as latest processed chunk
+                // TODO: add redis checkpoint save here with the latest saved wallet date
+            }
+            .doOnError {
+                logger.error(
```

Comment:
Avoid logging whole exception to limit risk of stacktrace being exposed

.doOnError { error ->
                logger.error(
                    "Error processing payment wallet chunk in time window: $startDate - $endDate. Error is: "+ error.message
                )
            }

Reply:
Same here, why we should not log stack trace? 

Why do you talk about expose stack trace here? those are application logs, stack trace is not put into any response body here.